### ucb patches to use system neon installed in non-standard directory; IssueZilla #53335
Index: ucb/source/ucp/webdav/LinkSequence.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/LinkSequence.cxx,v
retrieving revision 1.5
diff -u -r1.5 LinkSequence.cxx
--- ucb/source/ucp/webdav/LinkSequence.cxx	27 Jan 2005 12:12:09 -0000	1.5
+++ ucb/source/ucp/webdav/LinkSequence.cxx	14 Aug 2005 02:51:49 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _LINKSEQUENCE_HXX_
 #include "LinkSequence.hxx"
Index: ucb/source/ucp/webdav/LockEntrySequence.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/LockEntrySequence.cxx,v
retrieving revision 1.5
diff -u -r1.5 LockEntrySequence.cxx
--- ucb/source/ucp/webdav/LockEntrySequence.cxx	27 Jan 2005 12:12:40 -0000	1.5
+++ ucb/source/ucp/webdav/LockEntrySequence.cxx	14 Aug 2005 02:51:49 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _LOCKENTRYSEQUENCE_HXX_
 #include "LockEntrySequence.hxx"
Index: ucb/source/ucp/webdav/LockSequence.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/LockSequence.cxx,v
retrieving revision 1.6
diff -u -r1.6 LockSequence.cxx
--- ucb/source/ucp/webdav/LockSequence.cxx	27 Jan 2005 12:13:09 -0000	1.6
+++ ucb/source/ucp/webdav/LockSequence.cxx	14 Aug 2005 02:51:49 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _LOCKSEQUENCE_HXX_
 #include "LockSequence.hxx"
Index: ucb/source/ucp/webdav/NeonSession.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonSession.cxx,v
retrieving revision 1.34
diff -u -r1.34 NeonSession.cxx
--- ucb/source/ucp/webdav/NeonSession.cxx	27 Jan 2005 12:13:44 -0000	1.34
+++ ucb/source/ucp/webdav/NeonSession.cxx	14 Aug 2005 02:51:50 -0000
@@ -62,17 +62,33 @@
 #include <string.h>
 
 #ifndef NE_SOCKET_H
+#ifdef SYSTEM_NEON
+#include <ne_socket.h>
+#else
 #include <neon/ne_socket.h>
 #endif
+#endif
 #ifndef NE_AUTH_H
+#ifdef SYSTEM_NEON
+#include <ne_auth.h>
+#else
 #include <neon/ne_auth.h>
 #endif
+#endif
 #ifndef NE_REDIRECT_H
+#ifdef SYSTEM_NEON
+#include <ne_redirect.h>
+#else
 #include <neon/ne_redirect.h>
 #endif
+#endif
 #ifndef NE_LOCKS_H
+#ifdef SYSTEM_NEON
+#include <ne_locks.h>
+#else
 #include <neon/ne_locks.h>
 #endif
+#endif
 
 #ifndef _RTL_USTRBUF_HXX_
 #include <rtl/ustrbuf.hxx>
Index: ucb/source/ucp/webdav/NeonTypes.hxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonTypes.hxx,v
retrieving revision 1.9
diff -u -r1.9 NeonTypes.hxx
--- ucb/source/ucp/webdav/NeonTypes.hxx	27 Jan 2005 12:13:58 -0000	1.9
+++ ucb/source/ucp/webdav/NeonTypes.hxx	14 Aug 2005 02:51:50 -0000
@@ -63,17 +63,33 @@
 #define _NEONTYPES_HXX_
 
 #ifndef NE_SESSION_H
+#ifdef SYSTEM_NEON
+#include <ne_session.h>
+#else
 #include <neon/ne_session.h> // for ne_session
 #endif
+#endif
 #ifndef NE_UTILS_H
+#ifdef SYSTEM_NEON
+#include <ne_utils.h>
+#else
 #include <neon/ne_utils.h> // for ne_status
 #endif
+#endif
 #ifndef NE_BASIC_H
+#ifdef SYSTEM_NEON
+#include <ne_basic.h>
+#else
 #include <neon/ne_basic.h> // for ne_server_capabilities
 #endif
+#endif
 #ifndef NE_PROPS_H
+#ifdef SYSTEM_NEON
+#include <ne_props.h>
+#else
 #include <neon/ne_props.h> // for ne_propname, ne_prop_result_set
 #endif
+#endif
 
 typedef ne_session                  HttpSession;
 typedef ne_status                   HttpStatus;
Index: ucb/source/ucp/webdav/NeonUri.hxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonUri.hxx,v
retrieving revision 1.8
diff -u -r1.8 NeonUri.hxx
--- ucb/source/ucp/webdav/NeonUri.hxx	21 Aug 2002 07:34:53 -0000	1.8
+++ ucb/source/ucp/webdav/NeonUri.hxx	14 Aug 2005 02:51:50 -0000
@@ -62,8 +62,12 @@
 #define _NEONURI_HXX_
 
 #ifndef NE_URI_H
+#ifdef SYSTEM_NEON
+#include <ne_uri.h>
+#else
 #include <neon/ne_uri.h>
 #endif
+#endif
 
 #ifndef _RTL_USTRING_HXX_
 #include <rtl/ustring.hxx>
Index: ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx,v
retrieving revision 1.6
diff -u -r1.6 UCBDeadPropertyValue.cxx
--- ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx	27 Jan 2005 12:14:13 -0000	1.6
+++ ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx	14 Aug 2005 02:51:50 -0000
@@ -62,8 +62,12 @@
 #include <string.h>
 
 #ifndef NE_XML_H
+#ifdef SYSTEM_NEON
+#include <ne_xml.h>
+#else
 #include <neon/ne_xml.h>
 #endif
+#endif
 
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
Index: ucb/source/ucp/webdav/makefile.mk
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/makefile.mk,v
retrieving revision 1.23
diff -u -r1.23 makefile.mk
--- ucb/source/ucp/webdav/makefile.mk	27 Jan 2005 12:14:44 -0000	1.23
+++ ucb/source/ucp/webdav/makefile.mk	14 Aug 2005 02:51:50 -0000
@@ -85,6 +85,7 @@
 .INCLUDE: sv.mk
 
 .IF "$(SYSTEM_NEON)" == "YES"
+CFLAGS+= -DSYSTEM_NEON
 CFLAGS+= $(NEON_CFLAGS)
 .ELSE
 NEONINCDIR=external$/neon


### patch to use system jpeg; IssueZilla #53270
Index: solenv/inc/libs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/libs.mk,v
retrieving revision 1.92
diff -u -u -r1.92 libs.mk
--- solenv/inc/libs.mk	5 Aug 2005 15:04:49 -0000	1.92
+++ solenv/inc/libs.mk	15 Aug 2005 06:21:39 -0000
@@ -208,6 +208,8 @@
 #i34482# Blackdown jdk is in the libsearch patch and has a libjpeg
 .IF "$(OS)" == "FREEBSD"
 JPEG3RDLIB=/usr/local/lib/libjpeg.so
+.ELIF "$(OS)" == "MACOSX"
+JPEG3RDLIB=-ljpeg
 .ELSE
 JPEG3RDLIB=/usr/lib/libjpeg.so
 .ENDIF


### patch to use system python; IssueZilla #53280
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.129
diff -u -r1.129 configure.in
--- config_office/configure.in	5 Aug 2005 15:05:56 -0000	1.129
+++ config_office/configure.in	15 Aug 2005 08:14:48 -0000
@@ -2436,7 +2436,8 @@
    python_include=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('INCLUDEPY');"`
    python_version=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('VERSION');"`
    PYTHON_CFLAGS="-I$python_include"
-   PYTHON_LIBS="-lpython$python_version"
+   PYTHON_LIBS=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('LIBPL');"`
+   PYTHON_LIBS="-L$PYTHON_LIBS -lpython$python_version"
    
    dnl check if the headers really work:
    save_CPPFLAGS="$CPPFLAGS"


### patches for enabling the use of a scanner on Mac OSX; IssueZilla #48285
# Hunk fails; I will re-make a diff later.

### patches to detect locale correctly; IsuueZilla #25416
# http://www.openoffice.org/nonav/issues/showattachment.cgi/27507/sal_systools_macxp_extras_x11osx_osxlocale.c.patch
# http://www.openoffice.org/nonav/issues/showattachment.cgi/27508/sal_osl_unx_file.cxx.patch
# build failed in udkapi
# Already applied to m129 :(

### a patch to use system_boost; Not filed in IssueZilla :(
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.129
diff -u -r1.129 configure.in
--- config_office/configure.in	5 Aug 2005 15:05:56 -0000	1.129
+++ config_office/configure.in	16 Aug 2005 19:34:46 -0000
@@ -2010,7 +2010,7 @@
    AC_PROG_CPP
 fi
 dnl Check whether there's a C++ pre-processor.
-if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" ; then
+if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o "$_os" = "Darwin"; then
    AC_PROG_CXXCPP
 fi
 
@@ -2592,10 +2592,13 @@
 if test -n "$with_system_boost" && test "$with_system_boost" != "no"; then
    AC_MSG_RESULT([external])
    SYSTEM_BOOST=YES
+   AC_LANG_SAVE
+   AC_LANG_CPLUSPLUS
    AC_CHECK_HEADER(boost/shared_ptr.hpp, [],
        [AC_MSG_ERROR(boost/shared_ptr.hpp not found. install boost)], [])
    AC_CHECK_HEADER(boost/spirit/core.hpp, [],
        [AC_MSG_ERROR(boost/spirit/core.hpp not found. install boost)], [])
+   AC_LANG_RESTORE
 else
    AC_MSG_RESULT([internal])
    SYSTEM_BOOST=NO

### A patch to utilize Osaka font; IssueZilla #54535
--- officecfg/registry/data/org/openoffice/VCL.xcu	2005-09-08 23:46:30.000000000 +0900
+++ officecfg/registry/data/org/openoffice/VCL.xcu.fink	2005-10-17 03:08:38.000000000 +0900
@@ -434,31 +434,31 @@
     </node>
     <node oor:name="ja" oor:op="replace">
       <prop oor:name="CJK_TEXT" oor:type="xs:string" oor:op="replace">
-        <value>HG 明朝L Sun;HG MinchoL Sun;HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG Mincho Light J;MS P 明朝;TLP明朝;LX明朝;IPA P明朝;東風明朝;Kochi Mincho;さざなみ明朝;Mincho;Serif</value>
+        <value>HG 明朝L Sun;HG MinchoL Sun;HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG Mincho Light J;MS P 明朝;TLP明朝;LX明朝;IPA P明朝;Osaka;東風明朝;Kochi Mincho;さざなみ明朝;Mincho;Serif</value>
       </prop>
        <prop oor:name="CJK_HEADING" oor:type="xs:string" oor:op="replace">
-        <value>HG 明朝L Sun;HG 明朝L;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Gothic;MS 明朝;HG Mincho J;HG Mincho L;HG Mincho;Mincho;MS P 明朝;HG Mincho Light J;MS ゴシック;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS P ゴシック;Andale Sans UI</value>
+        <value>HG 明朝L Sun;HG 明朝L;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Gothic;MS 明朝;HG Mincho J;HG Mincho L;HG Mincho;Mincho;MS P 明朝;HG Mincho Light J;MS ゴシック;HG Gothic J;HG Gothic B;HG Gothic;Gothic;MS P ゴシック;Andale Sans UI</value>
       </prop>
       <prop oor:name="CJK_PRESENTATION" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;MS ゴシック;MS P ゴシック;HG Gothic;HG Gothic B;Gothic;Andale Sans UI</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;MS ゴシック;MS P ゴシック;HG Gothic;HG Gothic B;Gothic;Andale Sans UI</value>
       </prop>
       <prop oor:name="LATIN_PRESENTATION" oor:type="xs:string" oor:op="replace">
         <value>Andale;Arial</value>
       </prop>
       <prop oor:name="CJK_SPREADSHEET" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Andale Sans UI;Kochi Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Andale Sans UI;Kochi Gothic;HG Gothic J;HG Gothic B;HG Gothic;Gothic</value>
       </prop>
       <prop oor:name="LATIN_SPREADSHEET" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;Andale Sans UI;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Kochi Gothic;MS PGothic;Gothic</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;Andale Sans UI;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;MS Gothic;HG Gothic J;HG Gothic B;HG Gothic;Kochi Gothic;MS PGothic;Gothic</value>
       </prop>
       <prop oor:name="UI_FIXED" oor:type="xs:string" oor:op="replace">
-        <value>HG ゴシックB Sun;HG-GothicB Sun;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
+        <value>HG ゴシックB Sun;HG-GothicB Sun;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
        </prop>
        <prop oor:name="FIXED" oor:type="xs:string" oor:op="replace">
-        <value>HG ゴシックB Sun;HG-GothicB Sun;HG ゴシックB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
+        <value>HG ゴシックB Sun;HG-GothicB Sun;HG ゴシックB;HG Mincho Light J;MS P ゴシック;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic</value>
        </prop>
       <prop oor:name="UI_SANS" oor:type="xs:string" oor:op="replace">
-        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;Andale Sans UI;HG Mincho Light J;標準;TLPゴシック;LXゴシック;IPA Pゴシック;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic;Gnu-Unifont</value>
+        <value>HG PゴシックB Sun;HG-PGothicB Sun;HG PゴシックB;HG-PGothicB;HG-GothicB;Andale Sans UI;HG Mincho Light J;標準;TLPゴシック;LXゴシック;IPA Pゴシック;Osaka;東風ゴシック;さざなみゴシック;Kochi Gothic;Gothic;Gnu-Unifont</value>
       </prop>
       <prop oor:name="LATIN_FIXED" oor:type="xs:string" oor:op="replace">
         <value>hgmincholightj;cumberland;couriernew;courier;lucidasanstypewriter;lucidatypewriter;monaco;monospaced</value>
--- python/makefile.mk	Tue Jun  7 02:23:46 2005
+++ python/makefile.mk.fink	Tue Sep  6 22:40:22 2005
@@ -121,7 +121,7 @@
 BUILD_DIR=
 MYCWD=$(shell pwd)/$(INPATH)/misc/build
 CONFIGURE_ACTION= ./configure --prefix=$(MYCWD)/python-inst --enable-shared
-.IF "$(OS)$(CPU)" == "SOLARISI"
+.IF "$(OS)$(CPU)" == "SOLARISI" || "$(OS)$(CPU)" == "MACOSXP"
 CONFIGURE_ACTION += --disable-ipv6
 .ENDIF
 .IF "$(COMNAME)"=="sunpro5"
